local_units/management/commands/import-local-units-csv.py

```diff
@@ -19,57 +20,86 @@ def add_arguments(self, parser):
 
     @transaction.atomic
     def handle(self, *args, **options):
+        local_units_to_create = []
         filename = options['filename'][0]
         with open(filename) as csvfile:
             reader = csv.DictReader(csvfile)
             for i, row in enumerate(reader):
                 # Without positions we can't use the row:
                 if not row['LONGITUDE'] or not row['LATITUDE']:
+                    self.stdout.write(
+                        self.style.WARNING(f'Skipping row {i}: Empty locations point')
+                    )
                     continue
                 if len(row['POSTCODE']) > 10:
                     row['POSTCODE'] = ''  # better then inserting wrong textual data
-                unit = LocalUnit()
-                unit.country = Country.objects.get(iso3=row['ISO3'])
+                country = Country.objects.get(iso3=row['ISO3'])
                 # We do not check COUNTRY or NATIONAL_SOCIETY, but only this ^
-                if   row['TYPECODE'] == 'NS0': row['TYPECODE'] = 1
-                elif row['TYPECODE'] == 'NS1': row['TYPECODE'] = 2
-                elif row['TYPECODE'] == 'NS2': row['TYPECODE'] = 3
-                elif row['TYPECODE'] == 'NS3': row['TYPECODE'] = 4
-                else: row['TYPECODE'] = int(row['TYPECODE'])
-                unit.type = LocalUnitType.objects.get(
+                row['TYPECODE'] = {
+                    'NS0': 1,
+                    'NS1': 2,
+                    'NS2': 3,
+                    'NS3': 4,
+                }.get(
+                    row['TYPECODE'],
+                    int(row['TYPECODE']),
+                )
+                type = LocalUnitType.objects.get(
```

Comment:
Let's do the same for this as well.


Basically, we want to avoid unnecessary database query when we can. If the data is small, prefetching is fine.